src/http-client/util.ts

```diff
@@ -0,0 +1,68 @@
+import { HttpClientRequestOptions, TxnAuthJSON } from './interfaces';
+
+export function buildTxnAuthReqOptions(
+  txnAuthObj: TxnAuthJSON,
+  baseURL: string,
+  timeout: number,
+): HttpClientRequestOptions {
+  const advices = txnAuthObj.advices ? txnAuthObj.advices.TransactionConditionAdvice : [];
+  const transactionID = advices.reduce((prev: string, curr: string) => {
+    const prevWithSpace = prev ? ` ${prev}` : prev;
+    prev = `${curr}${prevWithSpace}`;
+    return prev;
+  }, '');
+
+  const openTags = `<Advices><AttributeValuePair>`;
+  const nameTag = `<Attribute name="TransactionConditionAdvice"/>`;
+  const valueTag = `<Value>${transactionID}</Value>`;
+  const endTags = `</AttributeValuePair></Advices>`;
+  const fullXML = `${openTags}${nameTag}${valueTag}${endTags}`;
+
+  const url = new URL(`${baseURL}json/realms/root/authenticate`);
+  url.searchParams.set('authIndexType', 'composite_advice');
+  url.searchParams.set('authIndexValue', fullXML);
+
+  const options = {
+    init: {
+      method: 'POST',
+    },
+    timeout,
+    url: url.toString(),
+  };
+  return options;
+}
+
+export async function examineForIGTxnAuth(res: Response): Promise<boolean> {
+  const type = res.headers.get('Content-Type') || '';
+  return Promise.resolve(type.includes('html') && res.url.includes('composite_advice'));
```

Comment:
In the same vein of keeping consistent with the top-level ergonomics of how these are used, I sided with `async` for both `examine*` and `normalize*` functions. This, of course, increases the runtime complexity a bit, but results in very clean implementations at the top-level. Let me know if this is okay.